bzt/modules/molotov.py

```diff
@@ -0,0 +1,187 @@
+        if load.hold:
+            hold = int(ceil(dehumanize_time(load.hold)))
+            duration += hold
+        cmdline += ['--duration', str(duration)]
+
+        cmdline += ['--use-extension=bzt.resources.molotov_ext']
+
+        cmdline += [self._get_script()]
+
+        env = {"MOLOTOV_TAURUS_REPORT": self.report_file_name}
+
+        self.start_time = time.time()
+        self.process = self.execute(cmdline, stdout=self.stdout_file, stderr=self.stderr_file, env=env)
+
+    def check(self):
+        ret_code = self.process.poll()
+        if ret_code is None:
+            return False
+        if ret_code != 0:
+            raise ToolError("molotov exited with non-zero code: %s" % ret_code, self.get_error_diagnostics())
+        return True
+
+    def shutdown(self):
+        shutdown_process(self.process, self.log)
+
+    def post_process(self):
+        if self.stdout_file and not self.stdout_file.closed:
+            self.stdout_file.close()
+        if self.stderr_file and not self.stderr_file.closed:
+            self.stderr_file.close()
+
+    def install_required_tools(self):
+        tool_path = self.settings.get('path', 'molotov')
+        tool = Molotov(tool_path, self.log)
+        if not tool.check_if_installed():
+            tool.install()
+        return tool_path
+
+    def get_error_diagnostics(self):
+        diagnostics = []
+        if self.stdout_file is not None:
+            with open(self.stdout_file.name) as fds:
+                contents = fds.read().strip()
+                if contents.strip():
+                    diagnostics.append("molotov STDOUT:\n" + contents)
+        if self.stderr_file is not None:
+            with open(self.stderr_file.name) as fds:
+                contents = fds.read().strip()
+                if contents.strip():
+                    diagnostics.append("molotov STDERR:\n" + contents)
+        return diagnostics
+
+    def resource_files(self):
+        return [self._get_script()]
+
+
+class Molotov(RequiredTool):
+    def __init__(self, tool_path, parent_logger):
+        super(Molotov, self).__init__("Molotov", tool_path)
+        self.tool_path = tool_path
+        self.log = parent_logger.getChild(self.__class__.__name__)
+
+    def check_if_installed(self):
+        self.log.debug('Checking Molotov: %s' % self.tool_path)
+        try:
+            stdout, stderr = communicate(shell_exec([self.tool_path, '--version']))
+            self.log.debug("Molotov stdout/stderr: %s, %s", stdout, stderr)
+            version_s = stdout.strip()
+            version = LooseVersion(version_s)
+            if version < LooseVersion("1.4"):
+                self.log.warning("Molotov version %s detected" % version)
+        except (CalledProcessError, OSError, AttributeError):
+            return False
+        return True
+
+    def install(self):
+        raise ToolError("You must install molotov tool (version 1.4 or greater) to use it")
+
+
+class MolotovReportReader(ResultsReader):
```

Comment:
If it's JMeter-like CSV, maybe just use JTL reader and that's it? Just like we do with Locust

Reply:
Format was changed to LDJSON to encapsulate more information, so it reuses `LDJSONReader` from utils.